### ISSUE TITLE
System.AccessViolationException - I can't mount drive

### ISSUE THREAD

**Issue by M0n7y5** [richardschneider/net-ipfs-mount#5]
Hello ... I've tried everything, but when i run `ipfs-mount -d y` it give me this error

```
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'ip4' (4).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'ip6' (41).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'tcp' (6).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'udp' (17).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'ipfs' (421).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'p2p' (420).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'http' (480).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'https' (443).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'dccp' (33).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'sctp' (132).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'ws' (477).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'libp2p-webrtc-star' (275).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'udt' (301).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'utp' (302).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'onion' (444).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'libp2p-webrtc-direct' (276).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'p2p-circuit' (290).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'dns' (53).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'dns4' (54).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'dns6' (55).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'dnsaddr' (12345).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'wss' (478).
12.08.2018 21:04:33 [DEBUG] Ipfs.NetworkProtocol - Registered 'ipcidr' (999).
12.08.2018 21:04:33 [DEBUG] Dokan - GetVolumeInformationProxy:
12.08.2018 21:04:33 [DEBUG] Dokan -     Context {<null>, False, False, False, False, #4, False, False}
12.08.2018 21:04:33 [DEBUG] Dokan -     VolumeNameBuffer        Interplanetary
12.08.2018 21:04:33 [DEBUG] Dokan -     FileSystemNameBuffer    IPFS
12.08.2018 21:04:33 [DEBUG] Dokan -     VolumeSerialNumber      41962596
12.08.2018 21:04:33 [DEBUG] Dokan -     FileSystemFlags CaseSensitiveSearch, CasePreservedNames, UnicodeOnDisk, PersistentAcls, SupportsRemoteStorage, SupportsObjectIDs, ReadOnlyVolume
12.08.2018 21:04:33 [DEBUG] Dokan - GetVolumeInformationProxy Return : Success
12.08.2018 21:04:33 [DEBUG] Dokan - MountedProxy:
12.08.2018 21:04:33 [DEBUG] Dokan -     Context {<null>, False, False, False, False, #0, False, False}
IPFS mounted
12.08.2018 21:04:33 [DEBUG] Dokan - MountedProxy Return : Success


Neošetřená výjimka: Neošetřená výjimka: System.AccessViolationException: Došlo k pokusu o čtení nebo zápis v chráněné paměti. Zpravidla se jedná o indikaci, že došlo k poškození další paměti.
   v DokanNet.Native.NativeMethods.DokanMapKernelToUserCreateFileFlags(UInt32 fileAttributes, UInt32 createOptions, UInt32 createDisposition, Int32& outFileAttributesAndFlags, Int32& outCreationDisposition)
   v DokanNet.DokanOperationProxy.ZwCreateFileProxy(String rawFileName, IntPtr securityContext, UInt32 rawDesiredAccess, UInt32 rawFileAttributes, UInt32 rawShareAccess, UInt32 rawCreateDisposition, UInt32 rawCreateOptions, DokanFileInfo rawFileInfo)
System.AccessViolationException: Došlo k pokusu o čtení nebo zápis v chráněné paměti. Zpravidla se jedná o indikaci, že došlo k poškození další paměti.
   v DokanNet.Native.NativeMethods.DokanMapKernelToUserCreateFileFlags(UInt32 fileAttributes, UInt32 createOptions, UInt32 createDisposition, Int32& outFileAttributesAndFlags, Int32& outCreationDisposition)
   v DokanNet.DokanOperationProxy.ZwCreateFileProxy(String rawFileName, IntPtr securityContext, UInt32 rawDesiredAccess, UInt32 rawFileAttributes, UInt32 rawShareAccess, UInt32 rawCreateDisposition, UInt32 rawCreateOptions, DokanFileInfo rawFileInfo)

Neošetřená výjimka: System.AccessViolationException: Došlo k pokusu o čtení nebo zápis v chráněné paměti. Zpravidla se jedná o indikaci, že došlo k poškození další paměti.
   v DokanNet.Native.NativeMethods.DokanMapKernelToUserCreateFileFlags(UInt32 fileAttributes, UInt32 createOptions, UInt32 createDisposition, Int32& outFileAttributesAndFlags, Int32& outCreationDisposition)
   v DokanNet.DokanOperationProxy.ZwCreateFileProxy(String rawFileName, IntPtr securityContext, UInt32 rawDesiredAccess, UInt32 rawFileAttributes, UInt32 rawShareAccess, UInt32 rawCreateDisposition, UInt32 rawCreateOptions, DokanFileInfo rawFileInfo)
```
Info:
net-ipfs-mount v0.2.0, v0.2.1, v0.2.2
Windows 10 Build 17134.191 (1803)

**Comment by richardschneider** [comment]
@M0n7y5 Thanks for the bug report. 

Dokany has released a new version (1.1.0.2000) that is causing this crash.  A temporary fix is to install dokany v1.0.1.1000.

- Uninstall dokany `choco uninstall dokany`
- Restart (reboot) the machine
- Install earlier dokany `choco install dokany --version 1.0.1.1000`

In the meantime I'll work on upgrading `net-ipfs-mount` to use latest `dokany`.

**Comment by richardschneider** [comment]
Fixed in [v0.3.0](https://github.com/richardschneider/net-ipfs-mount/releases/tag/v0.3.0).

Please note, I had some choco issues, so the latest release is v0.3.2
